MonoGame.Framework/Windows/WinFormsGameWindow.cs

```diff
@@ -195,7 +198,35 @@ internal WinFormsGameWindow(WinFormsGamePlatform platform)
 
             _form.KeyPress += OnKeyPress;
 
-            _allWindows.Add(this);
+            RegisterToAllWindows();
+        }
+
+        private void RegisterToAllWindows()
```

Comment:
@christallire - Can you submit the changes to `WinFormsGameWindow` in a separate PR?